packages/adapter-planetscale/src/conversion.ts

```diff
@@ -101,5 +101,12 @@ export const cast: typeof defaultCast = (field, value) => {
   if (field.type === 'JSON' && value === 'null') {
     return JsonNullMarker
   }
-  return defaultCast(field, value)
+
+  const defaultValue = defaultCast(field, value)
+
+  if (defaultValue instanceof Uint8Array) {
+    return Array.from(defaultValue)
+  }
```

Comment:
Why does it need to be turned into an array?

Reply:
Otherwise tests fail. A `Uint8Array` passed to Napi / Wasm would look like an `serde_json::Value::Object` shaped as:

```json
{
  "0": 21,
  "1": 25,
  "2": ...
}
```

Reply:
Applying the `if` above makes `PlanetScale` behave exactly like `Pg`.

Reply:
Hm... Can we somehow force napi to accept to `Uint8Array` in there and adjust neon as well? Transfering array buffer between JS and napi is really cheap (it basically amounts to transfering a pointer) where array requires a copy.

Reply:
Improving performance would indeed be better.
Would you be open to:
- merge this PR and its companion https://github.com/prisma/prisma-engines/pull/4695 now
- open an tech-debt issue to share `Uint8Array`s directly between `@prisma/client` and engines, which we can work on this Friday?

(I'm proposing this because [this issue](https://github.com/prisma/team-orm/issues/687) has been open for a while, and with this PR at least PlanetScale works with `Bytes`. Temporary closing this "byte correctness context" would also allow me to better focus on https://github.com/prisma/team-orm/issues/892).

Reply:
Yes, the plan is fine with me.

Reply:
Thanks. I've created https://github.com/prisma/team-orm/issues/926.